stl/inc/format

```diff
@@ -1496,6 +1496,214 @@ public:
+        return 0;
+    }
+
+    _NODISCARD size_t _Limit(const size_t _Size) noexcept {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    ptrdiff_t _Count_ = 0;
+    ptrdiff_t _Limit_;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(const ptrdiff_t _Limit) noexcept : _Limit_(_Limit) {}
+
+    _NODISCARD size_t _Count() const noexcept {
+        return static_cast<size_t>(_Count_);
+    }
+
+    _NODISCARD size_t _Limit(const size_t _Size) noexcept {
+        size_t _Avail = static_cast<size_t>(_Limit_ > _Count_ ? _Limit_ - _Count_ : 0);
+        _Count_ += _Size;
+        return _Size < _Avail ? _Size : _Avail;
+    }
+};
+
+inline constexpr size_t _Fmt_buffer_size = 256;
+
+template <class _OutputIt, class _Ty, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_Ty> {
+private:
+    _OutputIt _Output;
+    _Ty _Data[_Fmt_buffer_size];
+
+    void _Grow(size_t) final {
+        if (this->_Size() == _Fmt_buffer_size) {
+            _Flush();
+        }
+    }
+
+    void _Flush() {
+        auto _Size = this->_Size();
+        this->_Clear();
+        _Output = _STD _Copy_unchecked(_Data, _Data + this->_Limit(_Size), _STD move(_Output));
+    }
+
+public:
+    explicit _Fmt_iterator_buffer(_OutputIt _Out, ptrdiff_t _Size = _Fmt_buffer_size)
+        : _Traits(_Size), _Fmt_buffer<_Ty>(_Data, 0, _Fmt_buffer_size), _Output(_STD move(_Out)) {}
+
+    ~_Fmt_iterator_buffer() {
+        if (this->_Size() != 0) {
+            _Flush();
+        }
+    }
+
+    _NODISCARD _OutputIt _Out() {
+        _Flush();
+        return _STD move(_Output);
+    }
+
+    _NODISCARD ptrdiff_t _Count() const noexcept {
+        return static_cast<ptrdiff_t>(_Traits::_Count() + this->_Size());
+    }
+};
+
+template <class _Ty>
+class _Fmt_iterator_buffer<_Ty*, _Ty> final : public _Fmt_buffer<_Ty> {
+private:
+    void _Grow(size_t) final {}
+
+public:
+    explicit _Fmt_iterator_buffer(_Ty* _Out, ptrdiff_t = 0) : _Fmt_buffer<_Ty>(_Out, 0, ~size_t{}) {}
+
+    _NODISCARD _Ty* _Out() noexcept {
+        return this->end();
+    }
+};
+
+template <class _Container>
+class _Fmt_iterator_buffer<back_insert_iterator<_Container>,
+    enable_if_t<contiguous_iterator<typename _Container::iterator>, typename _Container::value_type>>
+    final : public _Fmt_buffer<typename _Container::value_type> {
+private:
+    _Container& _Cont;
+
+    struct _Accessor : back_insert_iterator<_Container> {
+        explicit _Accessor(back_insert_iterator<_Container> _Iter) : back_insert_iterator<_Container>(_Iter) {}
+
+        using back_insert_iterator<_Container>::container;
+    };
+
+    void _Grow(size_t _Capacity) final {
+        _Cont.resize(_Capacity);
+        this->_Set(_RANGES data(_Cont), _Capacity);
+    }
+
+public:
+    explicit _Fmt_iterator_buffer(_Container& _Cont_)
+        : _Fmt_buffer<typename _Container::value_type>(_Cont_.size()), _Cont(_Cont_) {}

```

Comment:
Must use `_RANGES size` and `range_value_t` (`_Container::value_type` doesn't have to be defined or might mean a completely different thing)
```suggestion
        : _Fmt_buffer<_RANGES range_value_t<_Container>>(_RANGES size(_Cont_)), _Cont(_Cont_) {}
```